src/game.cpp

```diff
@@ -803,13 +803,30 @@ class GameOnDemandSoundFetcher: public OnDemandSoundFetcher
 	}
 };
 
+// before 1.8 there isn't a "integer interface", only float
+#if (IRRLICHT_VERSION_MAJOR == 1 && IRRLICHT_VERSION_MINOR < 8)
+typedef irr::f32 samplerLayer_t;
+#else
+typedef irr::s32 SamplerLayer_t;
+#endif
```

Comment:
1. No, not intentional. I'll fix this.
2. No, I'm not. I'm conditionally defining SamplerLayer_t.